platform-coop-toolkit/coop-library-framework
# Add date fields (resolves #106)

## Description

This PR replaces publication date and year fields with publication year, month, and day fields. Month and day fields are only enabled if a valid year has been entered. Changing the year or month field's value rebuilds the select control which lets the user choose the publication day, preserving an existing day if it is present in the newly chosen year/month. For example:

- If the publication date is set to February 29, 2016 (a leap year) and the year is changed to 2012, the day will be preserved (new date: February 29, 2012).
- If the publication date is set to February 29, 2016 and the month is changed to January, the day will be preserved (new date: January 29, 2016).
- If the publication date is set to February 29, 2016 and the year is changed to 2019, the day will be reset as February 29, 2019 is not a valid date.

## Steps to test

1. Add a year, month, and day.
2. Save.
3. Replace both the month and year, and verify that the date selection fields still reflect the correct number of days in the given month/year combination.

**Expected behavior:** Years in the future or distant past are flagged as invalid. Changing year or month adjusts the day field to reflect the relevant choices for the given year and month, preserving user input wherever possible.

## Additional information

Not applicable.

## Related issues

- Resolves #106
